### PR TITLE
Add .git remote data for GitHub, Bitbucket Cloud and Azure Repos + Fix Azure Repos download repo bug

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---
* Add .git remote data for GitHub download repository 
* Add .git remote data for Azure Repos download repository 
* Add .git remote data for Bitbucket Cloud Download Repository
* Fix Azure Repos download repo bug

### CODE REVIEW

File: vcsutils/utils_test.go [hunk 1]
```diff
@@ -148,3 +148,21 @@ func TestGetZeroValue(t *testing.T) {
...
+
+func TestCreateDotGitFolderWithRemote(t *testing.T) {
+	dir1, err := os.MkdirTemp("", "tmp")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dir1)
+	err = CreateDotGitFolderWithRemote(dir1, "origin", "fakeurl")
+	assert.NoError(t, err)
+	dotGitExist, err := os.Stat(filepath.Join(dir1, ".git"))
+	assert.NoError(t, err)
+	assert.Equal(t, dotGitExist.Name(), ".git")
```
Comment (yahavi) [inline vcsutils/utils_test.go:160]: Let's add validation for the remote. Something like this:
```go
repo, err := git.PlainOpen(filepath.Join(dir1, ".git"))
assert.NoError(t, err)
remote, err := repo.Remote("origin")
assert.NotNil(t, remote)
assert.Contains(t, remote.Config().URLs, "fakeurl")
```

File: vcsutils/utils_test.go [hunk 2]
```diff
@@ -148,3 +148,21 @@ func TestGetZeroValue(t *testing.T) {
...
+	// Return error if .git already exist
+	assert.Error(t, CreateDotGitFolderWithRemote(dir1, "origin", "fakeurl"))
+	// Return error if remote name is empty
+	dir2, err := os.MkdirTemp("", "tmp")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dir2)
+	assert.Error(t, CreateDotGitFolderWithRemote(dir2, "", "fakeurl"))
```
Comment (yahavi) [inline vcsutils/utils_test.go:167]: I think it is worth putting these lines in a new unit test: TestCreateDotGitFolderWithoutRemote

File: vcsutils/utils.go [hunk 3]
```diff
@@ -224,3 +231,15 @@ func unzipFile(f *zip.File, destination string) (err error) {
 	}()
 	return safeCopy(destinationFile, zippedFile)
 }
+
+func CreateDotGitFolderWithRemote(path, remoteName, remoteUrl string) error {
```
Comment (yahavi) [inline vcsutils/utils.go:235]: Let's stick to the convention to put documentation above all public functions
```suggestion
// CreateDotGitFolderWithRemote ...
func CreateDotGitFolderWithRemote(path, remoteName, remoteUrl string) error {
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: vcsutils/utils_test.go (hunk 1 above)
Comment (omerzi) [reply vcsutils/utils_test.go:160]: done

File: vcsutils/utils.go (hunk 3 above)
Comment (omerzi) [reply vcsutils/utils.go:235]: done

File: vcsclient/azurerepos_test.go [hunk 4]
```diff
@@ -98,6 +98,24 @@ func TestAzureRepos_TestDownloadRepository(t *testing.T) {
...
+		}
+		if file.Name() == "README.md" {
+			readmeFound = true
+		}
+		if file.Name() == ".git" {
+			dotGitFound = true
+		}
+	}
+	assert.True(t, readmeFound)
+	assert.True(t, dotGitFound)
```
Comment (yahavi) [inline vcsclient/azurerepos_test.go:118]: Please consider this:
```suggestion
	assert.FileExists(t, filepath.Join(dir, "README.md"))
	assert.DirExists(t, filepath.Join(dir, ".git"))
```

File: vcsclient/bitbucketcloud_test.go [hunk 5]
```diff
@@ -145,12 +145,20 @@ func TestBitbucketCloud_DownloadRepository(t *testing.T) {
...
+		}
 		if file.Name() == "README.md" {
 			readmeFound = true
 		}
+		if file.Name() == ".git" {
+			dotGitFound = true
+		}
 	}
 	assert.True(t, readmeFound)
+	assert.True(t, dotGitFound)
```
Comment (yahavi) [inline vcsclient/bitbucketcloud_test.go:161]:
```suggestion
	assert.FileExists(t, filepath.Join(dir, "README.md"))
	assert.DirExists(t, filepath.Join(dir, ".git"))
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: vcsclient/azurerepos_test.go (hunk 4 above)
Comment (omerzi) [reply vcsclient/azurerepos_test.go:118]: done

File: vcsclient/bitbucketcloud_test.go (hunk 5 above)
Comment (omerzi) [reply vcsclient/bitbucketcloud_test.go:161]: done

**Pull request merged**